integrations/docker/images/chip-build/Dockerfile

```diff
@@ -116,3 +116,6 @@ RUN set -x \
     && cd .. \
     && rm -rf node_js \
     && : # last line
+
+ENV NPM_CONFIG_PREFIX=/home/node/.npm-global
```

Comment:
These paths do not appear to exist in this container by default. 
```
[rwalker@anodyne chip]$ ./integrations/docker/images/chip-build/run.sh ls -a /home/node
```
`/home/node` doesn't seem to be present in the vscode devcontainer, either...

How did you test this change?

Reply:
@rwalker-apple My test process was the following. 
1. Compile locally a `chip-build` container
2. Create a `docker-compose` file that link my local repo of CHIP in the docker container.
3. Create a bash scripts which call `npm CI` and other setup command before running the `zap_regen_all.py`
4. Delete the `/third_party/zap/repo/node_modules` folder to remove all modules that where installed previously. That we can truly know if the fix works since the docker container will have to re-install those dependency before generating the files.
5. Run the the script created at step 3 in the CHIP-build container. It fails with `Error: EACCES: permission denied,...` The same behaviour was already observed when I tried to generated the templates with GitHub CI.
6. Add the fix to the DockerFile and rebuild the container.
7. Re-run the script created at step 3. All templates are successfully generated without any errors.

Reply:
I also wonder who creates the `node` user in the container. Do you know at what step this user gets created? How did you figure out that these prefixes/path changes are the required ones?

based on the nodejs install above for example, I would expect node binaries too exist in one of `/opt/node-v12.19.0-linux-x64/bin` _or_ `/opt/node/bin` _or_ `/usr/bin`

Reply:
I see https://github.com/nodejs/docker-node/blob/master/docs/BestPractices.md#non-root-user says that node images have such a non-root user (the node one).

This seems to require the docker image to derive from node but we derive from ubuntu:focal

Reply:
https://github.com/nodejs/docker-node/blob/ad676318f09f9dee821ac6704340000d75fe31bc/15/stretch/Dockerfile#L3 seems to do a 'useradd'.

We could do the same here and ensure a user is created.
I am still unsure what user CI uses. Since we define none, I would imagine 'root' (which sould explain rights to just use some non-existing /home dir) ... but then why would root get EACCESS.

Could we run a `whoami` before we get the EACCESS to understand the error?

Reply:
@andy31415 sorry for the delay, I can confirm that the user is indeed root in the build containers. While creating a non-root user is the best practice I don't think it will fix the problem with the global dependencies. So we may have to implement both. On that note, this fix works with a root user on my local `chip-build` container...

Reply:
My complain is that this seems to reference a user home directory that does not exist. I would argue that this needs more work because of that - even if the change seems to make chip-build still work (I imagine it does because root can do anything, including clobbering any /home directory).

I would like to have some solution that explains the problem (I am still unclear how we get EACCESS when running as root ... that makes no sense) and how we solve it (referencing a home directory of a user that does not exist sounds wrong to me)